federation/pkg/federation-controller/secret/secret_controller.go

```diff
@@ -277,8 +279,8 @@ func (secretcontroller *SecretController) Run(stopChan <-chan struct{}) {
 }
 
 func (secretcontroller *SecretController) deliverSecretObj(obj interface{}, delay time.Duration, failed bool) {
-	secret := obj.(*apiv1.Secret)
-	secretcontroller.deliverSecret(types.NamespacedName{Namespace: secret.Namespace, Name: secret.Name}, delay, failed)
+	namespacedName := secretcontroller.adapter.NamespacedName(obj.(pkgruntime.Object))
```

Comment:
I think you can modify the obj's argument type to pkgruntime.Object: it looks like both callers have that much type info.